src/schedule/zephyr_dp_schedule.c

```diff
@@ -375,6 +394,29 @@ static int scheduler_dp_task_shedule(void *data, struct task *task, uint64_t sta
 		return -EINVAL;
 	}
 
+	/* create a zephyr thread for the task */
+	thread_id = k_thread_create(&pdata->thread, (__sparse_force void *)pdata->p_stack,
+				    pdata->stack_size, dp_thread_fn, task, NULL, NULL,
+				    ZEPHYR_DP_THREAD_PRIORITY, K_USER, K_FOREVER);
+	if (!thread_id)	{
```

Comment:
this won't check if the function has failed, in fact it never fails

Reply:
according to zephyr docs - you're right. removing the check